src/main/java/abstracts/Car.java

```
@@ -0,0 +1,217 @@
+package abstracts;
+
+import interfaces.*;
+import org.apache.log4j.Logger;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+public abstract class Car implements Vehicle {
+
+    /**
+     * log4j var for logging
+     */
+    final static Logger logger = Logger.getLogger(Car.class);
+
+
+    /**
+     * Attributes for Car
+     */
+    private float fuel;
+    private int numberOfGears;
+    private int actualGear = 0;
+    private float loadCapacity;
+    private int numberOfDoors;
+    private int maxNumberOfOccupants;
+    private Set<String> occupantsNames = Collections.emptySet();
+
+
+    //Implementation for vehicle interface
+    public void accelerateVehicle() {
+        System.out.println("Vrum");
+    }
+
+    public void breakVehicle() {
+        System.out.println("Break");
+    }
+
+    public void turnLeft() {
+        System.out.println("Turning Left");
+    }
+
+    public void turnRight() {
+        System.out.println("Turning Right");
+    }
+
+    public float checkFuel() {
+
+        return fuel;
+    }
+
+    public void turnVehicleOn() {
```

There should be at least a boolean field to hold whether the vehicle is turned on or not. If we try to turn on an already turned on vehicle it should throw an exception.